sdk/trace/snapshot.go

```diff
@@ -104,7 +104,7 @@ func (s snapshot) Status() Status {
 
 // InstrumentationLibrary returns information about the instrumentation
 // library that created the span.
-func (s snapshot) InstrumentationLibrary() instrumentation.Library {
+func (s snapshot) InstrumentationLibrary() instrumentation.Scope {
```

Comment:
This introduces an incompatible change to a stable module, something we cannot do according to our stability and versioning [policy](https://github.com/open-telemetry/opentelemetry-go/blob/main/VERSIONING.md#policy).

Instead, this method (and the other methods on exported types) should remain but itself be deprecated. A replacement `InstrumentationScope` method can be added for users to use instead.

Reply:
Thanks! I updated the PR, but wasn't quite sure how far to go. I think a lot of the internal uses are due to the `sdk.metric.export.InstrumentationLibraryReader` interface. I suppose I could try to deprecate it and create an `InstrumentationScopeReader` interface? Also `sdk.trace.tracetest.SpanStub` contains a `InstrumentationLibrary`. Should I deprecate that and add an `InstrumentationScope instrumentation.Scope`?


Thank you!